bcipy/signal/model/pca_rda_kde/pca_rda_kde.py

```diff
@@ -0,0 +1,113 @@
+import pickle
+from pathlib import Path
+from typing import List
+
+import numpy as np
+from ..base_model import ModelEvaluationReport, SignalModel
+from .classifier import RegularizedDiscriminantAnalysis
+from .cross_validation import cost_cross_validation_auc, cross_validation
+from .density_estimation import KernelDensityEstimate
+from .dimensionality_reduction import ChannelWisePrincipalComponentAnalysis
+from .pipeline import Pipeline
+from scipy.stats import iqr
+
+
+class NotFittedError(Exception):
+    pass
+
+
+class PcaRdaKdeModel(SignalModel):
+    def __init__(self, k_folds=10):
+        self.k_folds = k_folds
+        self.model = None
+        self._ready_to_predict = False
+
+    def fit(self, train_data: np.array, train_labels: np.array) -> SignalModel:
+        """
+        Train on provided data using K-fold cross validation and return self.
+
+        Parameters:
+            train_data: shape (Channels, Trials, Trial_length) preprocessed data
+            train_labels: shape (Trials,) binary labels
+
+        Returns:
+            trained likelihood model
+        """
+        pca = ChannelWisePrincipalComponentAnalysis(var_tol=1e-5, num_ch=train_data.shape[0])
+        rda = RegularizedDiscriminantAnalysis()
+        model = Pipeline()
```

Comment:
As a future improvement it might be nice for `Pipeline` to be able to handle *args: `model = Pipeline(pca, rda)`.

Reply:
added